google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/FlexibleFacetEditor.java

```diff
@@ -287,19 +306,35 @@ public void onFacetInitialized(@NotNull Facet facet) {
         Project project,
         String fileName,
         Consumer<Path> configFileGenerator,
-        TextFieldWithBrowseButton filePicker,
+        TextFieldWithBrowseButton directoryPicker,
         Runnable configurationValidator) {
       this.project = project;
       this.fileName = fileName;
       this.configFileGenerator = configFileGenerator;
-      this.filePicker = filePicker;
+      this.directoryPicker = directoryPicker;
       this.configurationValidator = configurationValidator;
     }
 
     @Override
     public void actionPerformed(ActionEvent event) {
+      boolean genAppYaml = "app.yaml".equals(fileName);
```

Comment:
this feels a bit off. this is supposed to be a generic generator yet you add app.yaml specific stuff here. I'm thinking this can be refactored to avoid this.